### PR TITLE
Feature/ui improvements

### LINKED ISSUE #82: Tracking: feature/ui-improvements
Branch Tracking Issue for feature/ui-improvements
=================================
**Overview:**
This branch will focus on various usability improvements to the DGP user interface. Improving user interaction with the plotters, updating icons/visual elements, integrating work on user settings (persistent application state), etc.

**Goals:**
- [x] Implement more user friendly plot controls for plotting data channels on different plots/axis' in line selection plot
- [x] Integrate User Settings features, e.g. saving/restoring window position, improving recent-projects tracking/loading, restoring tabs to their last state (plotted lines etc)
- [x] Add ability to hide/show segment selections on the LineSelectPlot
- [x] Refactor workspace tab interface, create specific tab classes for different controllers (i.e. Project/Flight/DataSet).
- [x] Add basic workspace tab state persistence, i.e. the plot tab will restore whichever series were plotted when it was last closed.

**Future Branch:**
- [ ] Implement custom UI Dialog for displaying formatted text e.g. for in-application help, graph-transform source etc. Similar to a QMessageDialog but with customizable central widget e.g. a QTextEdit or QTextBrowser
- [ ] Add option menu to plots enabling user to select the number of rows/columns (within limits) of plots to display. (may need to trigger a complete reload of the tab).

### PR DESCRIPTION
**Pull request by bradyzp**
UI Improvements Branch
================
The UI Improvements branch addresses various quality-of-life improvements in the user interface.
- Adds support for storing user settings, better tracking of recent projects, restores main-window state to last known state when launching.
- Implements more intuitive plotting controls for the segment selection plot
- Redesigns the workspace/tab-interface to be more extensible in the future, also adds state save/restore functionality so that individual tabs can load to their last state (i.e. plot what the user had plotted when the tab was closed)
- Adds new Icons and cleans up the resource file. New icons are sourced from [Material.io](https://material.io/tools/icons) which provides open-source icon files with a uniform appearance.
- Adds a documentation launcher action to the application, taking user directly to the DGP docs page on read-the-docs.
- Add initial support to link gravimeter/sensor configuration to specific data-set, w/ visual display in the project tree.

Related Issues
---------------
Closes #82

### CODE REVIEW

File: dgp/gui/settings.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    def model(self):
+        return self._model
+
+    @property
+    def project_refs(self) -> Generator[RecentProject, None, None]:
+        for i in range(self.model.rowCount()):
+            yield self.model.item(i).data(RefRole)
+
+    def last_project_path(self) -> MaybePath:
+        path = Path(self._settings.value(SettingsKey.LastProjectPath(), None))
```
Comment (cbertinato) [inline dgp/gui/settings.py]: `Path` doesn't take `None` as an argument.  Testing without any prior projects raises an exception.

**Review by cbertinato — CHANGES_REQUESTED** [review]

**Comment by cbertinato** [comment]
Also, a couple of issues that came up in QA testing:
- The new splash screen does not disappear for me when the recent project dialog comes up. 
screen shot 2018-08-26 at 10 07 30 am

- In the project tree, if no datasets have been loaded for a flight, and if I double-click 'No Data', then an exception is raised:

```
Traceback (most recent call last):
  File "/Users/cbertinato/Documents/Git/DGP/dgp/gui/views/project_tree_view.py", line 78, in _on_double_click
    self.model().item_activated(index)
  File "/Users/cbertinato/Documents/Git/DGP/dgp/core/controllers/project_treemodel.py", line 127, in item_activated
    self.tabOpenRequested.emit(item.uid, item)
  File "/Users/cbertinato/Documents/Git/DGP/dgp/core/controllers/datafile_controller.py", line 32, in uid
    return self._datafile.uid
AttributeError: 'NoneType' object has no attribute 'uid'
```
screen shot 2018-08-26 at 10 11 23 am

**Comment by cbertinato** [comment]
Before adding a data to a dataset I had changed the name of the dataset from the default value.  Then proceeded to add a trajectory.  The old name of the dataset appeared in the dialog.  The import still went through with no issue.
screen shot 2018-08-26 at 10 28 59 am

**Comment by cbertinato** [comment]
When I minimize the windows (in Mac OSX), and then maximize, then the project tree side bar is toggled off closed.  I can bring it back with the toggle button on top.  Not sure if this is an issue in Windows.

**Comment by cbertinato** [comment]
Also, I can't seem to get the fields from the gravity data to appear in channels boxes on the right of the plots.  The issue appears also to happen for trajectory, but at some point the lat, lon, and height fields did appear, though I don't recall what series of actions I made for that to happen.

**Comment by bradyzp** [comment]
Hey Chris, thanks for doing some QC and reviewing!

I've pushed up a change set which should resolve the Path issue in settings.py, the exception caused by trying to activate/double-click empty DataFiles, and the issue where the project tree/dock is being hidden after minimizing the window. Also the splash screen visibility issue should have been resolved.

The issue you point out where the DataSet name is not being updated in the import dialog, and the issue where you're not seeing the data channels appear are sort of related I think.
- The DataSet name isn't being updated correctly because the DataSetController is being cloned for display in the import dialog. (And the clone isn't being updated yet)
- Currently if you have a DataSet tab (with the plots) open, and then import a datafile into that dataset, there is no notification to the tab that new data has become available. If you close and reopen the tab you should see the channels.

Both of these issues are sort of related, in that there needs to be some sort of communication between the controllers and 1. any clones that they have, and 2. any UI items such as Tabs that their state has changed.
I've started working on this in a separate branch as I didn't want to pollute this one with too many changes, but basically my plan is to create a standard interface for all controllers which enables them to track (by weak-reference) any clones or references to themselves, and can then notify any referent objects of a state change.

**Review by cbertinato — APPROVED** [review]

**Pull request merged**